test/algconvergence/non-autonomous_convergence_tests.jl

```diff
@@ -0,0 +1,73 @@
+using OrdinaryDiffEq, DiffEqDevTools, Test
+
+function nonauto1(u, p, t)
+  x, _ = u
+  [
+    t*x
+    0
+  ]
+end
+
+function nonauto2(u, p, t)
+  _, y = u
+  [
+   y,
+   t*y
+  ]
+end
+
+function analytic(u0,p,t)
+  x0, y0 = u0
+  et = exp(t^2/2)
+  [
+   et*(x0 + t*y0),
+   et*y0
+  ]
+end
+
+u0 = [1.1, 2.2]
+tspan = (0.0,1.0)
+prob1 = ODEProblem(ODEFunction{true}((du,u,p,t)->du .= nonauto1(u,p,t) .+ nonauto2(u,p,t),
+                              analytic=analytic),
+                              u0,tspan)
+prob2 = ODEProblem(ODEFunction{false}((u,p,t)-> nonauto1(u,p,t) .+ nonauto2(u,p,t),
+                              analytic=analytic),
+                              u0,tspan)
+prob3 = SplitODEProblem(SplitFunction{true}((du,u,p,t)->du .= nonauto1(u,p,t),
+                                       (du,u,p,t)->du .= nonauto2(u,p,t),
+                              analytic=analytic),
+                              u0,tspan)
+prob4 = SplitODEProblem(SplitFunction{false}(nonauto1,
+                                        nonauto2,
+                              analytic=analytic),
+                              u0,tspan)
+
+testTol = 0.2
+
+kwargs = (reltol=1e-16, abstol=1e-16)
+for prob in [prob1, prob2, prob3, prob4]
+  dts = 1 .//2 .^(12:-1:8)
+  sim = test_convergence(dts,prob,KenCarp3(); kwargs...)
```

Comment:
is it not possible to get it passing without this?

Reply:
Yes, it is necessary.

Reply:
no changes to the nlsolver could do it?

Reply:
It must need the tolerance information to converge to the correct accuracy. I don't see there is anything that nlsolver could do.

Reply:
Oh yes, I see what you mean. It needs the tolerance so that when the fixed dt is small it actually converges enough. Interesting.